code/campaign/src/main/java/com/adobe/marketing/mobile/campaign/LocalNotificationHandler.java

```diff
@@ -0,0 +1,286 @@
+        if (message == null) {
+            Log.debug(LOG_TAG, SELF_TAG, "%s (local notification message)", Log.UNEXPECTED_NULL_VALUE);
+            return;
+        }
+
+        Activity currentActivity =
+                ServiceProvider.getInstance().getAppContextService().getCurrentActivity();
+        Intent resumeIntent;
+
+        // if we have a deep link, we need to create a new Intent because the old intents are using
+        // setClass (overrides opening a deeplink)
+        if (deeplink != null && !deeplink.isEmpty()) {
+            resumeIntent = new Intent(Intent.ACTION_VIEW);
+            resumeIntent.setData(Uri.parse(deeplink));
+        } else if (currentActivity != null) {
+            resumeIntent = currentActivity.getIntent();
+        } else {
+            resumeIntent = intent;
+        }
+
+        resumeIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
+        resumeIntent.putExtra(NOTIFICATION_IDENTIFIER_KEY, messageID);
+        resumeIntent.putExtra(NOTIFICATION_USER_INFO_KEY, userInfo);
+
+        final int buildVersion = Build.VERSION.SDK_INT;
+        NotificationManager notificationManager =
+                (NotificationManager) appContext.getSystemService(Context.NOTIFICATION_SERVICE);
+
+        try {
+            // if we have an activity for this notification, use it
+            PendingIntent sender;
+
+            if (buildVersion >= Build.VERSION_CODES.M) {
+                sender =
+                        PendingIntent.getActivity(
+                                appContext,
+                                senderCode,
+                                resumeIntent,
+                                PendingIntent.FLAG_UPDATE_CURRENT | PendingIntent.FLAG_IMMUTABLE);
+            } else {
+                sender =
+                        PendingIntent.getActivity(
+                                appContext,
+                                senderCode,
+                                resumeIntent,
+                                PendingIntent.FLAG_UPDATE_CURRENT);
+            }
+
+            if (sender == null) {
+                Log.debug(LOG_TAG,
+                        SELF_TAG,
+                        "Failed to retrieve sender from broadcast, unable to post notification");
+                return;
+            }
+
+            // Todo: This seems redundant as the App is first launched before handling Broadcast
+            // intent
+            // App.INSTANCE.setAppContext(context.getApplicationContext());
+            DeviceInforming systemInfoService =
+                    ServiceProvider.getInstance().getDeviceInfoService();
+            String appName = systemInfoService.getApplicationName();
+
+            // notification channels are required if api level is 26 or higher
+            if (buildVersion >= Build.VERSION_CODES.O) {
+                final NotificationChannel notificationChannel = new NotificationChannel(NOTIFICATION_CHANNEL_ID,
```

Comment:
we should add code to check if the notification channel was already created previously prior to creating a new channel and adding it to the NotificationManager